### PR TITLE
Release v0.20.0

### PR DESCRIPTION
**Pull request by afeiszli**
## Checklist before requesting a review
- [ ] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [ ] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

**Review by mattkasun — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]

File: compose/docker-compose-emqx.yml [hunk 1]
```diff
@@ -1,85 +1,17 @@
...
   mq:
     container_name: mq
     image: emqx/emqx:5.0.9
     restart: unless-stopped
     environment:
-      EMQX_NAME: "emqx"
-      EMQX_DASHBOARD__DEFAULT_PASSWORD: "REPLACE_MQ_PASSWORD"
-      EMQX_DASHBOARD__DEFAULT_USERNAME: "REPLACE_MQ_USERNAME"
+      - EMQX_NAME: "emqx"
+      - EMQX_DASHBOARD__DEFAULT_PASSWORD=${MQ_PASSWORD}
```
Comment (gabrielseibel1) [inline compose/docker-compose-emqx.yml:10]: is it ok to use `=` here?

File: compose/docker-compose.ee.yml [hunk 2]
```diff
@@ -1,108 +1,29 @@
...
+
   prometheus:
     container_name: prometheus
     image: gravitl/netmaker-prometheus:latest
+    env_file: ./netmaker.env
     environment:
-      NETMAKER_METRICS_TARGET: "netmaker-exporter.NETMAKER_BASE_DOMAIN"
-      LICENSE_KEY: "YOUR_LICENSE_KEY"
+      # config-dependant vars
+      - NETMAKER_METRICS_TARGET=netmaker-exporter.${NM_DOMAIN}
```
Comment (gabrielseibel1) [inline compose/docker-compose.ee.yml:11]: ok to use `=`?

File: controllers/ext_client.go [hunk 3]
```diff
@@ -648,3 +643,35 @@ func doesUserOwnClient(username, clientID, network string) (bool, bool) {
 
 	return false, logic.StringSliceContains(netUser.Clients, clientID)
 }
+
+// validateExtClient	Validates the extclient object
+func validateExtClient(extclient *models.ExtClient, customExtClient *models.CustomExtClient) error {
+	//validate clientid
+	if customExtClient.ClientID != "" && !validName(customExtClient.ClientID) {
```
Comment (gabrielseibel1) [inline controllers/ext_client.go:650]: `if customExtClient.ClientID == "" || !validName(customExtClient.ClientID)`?
if the id is empty or is not valid

**Review by gabrielseibel1 — CHANGES_REQUESTED** [review]

File: controllers/ext_client.go (hunk 3 above)
Comment (mattkasun) [reply controllers/ext_client.go:650]: only validate if a client name was provided.  it is optional

**Comment by luzfcb** [comment]
I was installing by following the documentation (https://docs.netmaker.org/quick-start.html#install-netmaker ) because I want to learn step-by-step everything needed for Netmaker to work, and I noticed that the tutorial had not been updated  with the changes of this pull-request

**Comment by luzfcb** [comment]
Hello, I was installing by following the documentation (https://docs.netmaker.org/quick-start.html#install-netmaker ) because I want to learn step-by-step everything needed for Netmaker to work, and I noticed that the tutorial had not been updated with the changes of this pull-request.

The link to the NM-QUICK-INTERACTIVE SCRIPT is also outdated. in the documentation

**Pull request merged**